anchors/chain_repository.go

```diff
@@ -99,8 +121,33 @@ func (r repository) Commit(
 		types.NewHash(proof[:]),
 		types.NewMoment(storedUntil))
 	if err != nil {
-		return txHash, bn, sig, err
+		return nil, err
+	}
+
+	did, err := getDID(ctx)
+	if err != nil {
+		return nil, err
 	}
 
-	return r.api.SubmitExtrinsic(meta, c, krp)
+	jobID := contextutil.Job(ctx)
+	cctx := contextutil.Copy(ctx)
+	_, done, err := r.jobsMan.ExecuteWithinJob(cctx, did, jobID, "Check Job for anchor commit", r.api.SubmitAndWatch(cctx, meta, c, krp))
+
+	return done, err
+}
+
+// AnchorData holds data returned from previously anchored data against centchain
+type AnchorData struct {
+	AnchorID     types.Hash `json:"id"`
+	DocumentRoot types.Hash `json:"doc_root"`
+	BlockNumber  uint32     `json:"anchored_block"`
+}
+
+func (r repository) GetAnchorByID(id *big.Int) (*AnchorData, error) {
+	var ad *AnchorData
```

Comment:
is this a pointer to AnchorData?
We seem to pass the double pointer to the api.Call
Is that expected ?

Reply:
Seems that it needs a pointer of course and looks like the currently defined seems redundant, but it doesnt work unless I pass with `&` I made a change to only define it by value and then pass by reference in my next commit. 
Thanks for catching it